tests/e2e/kserve_test.go

```diff
@@ -511,3 +515,146 @@ func (tc *KserveTestCtx) createConnectionSecret(secretName, namespace string) {
+		WithMinimalObject(gvk.DataScienceCluster, tc.DataScienceClusterNamespacedName),
+		WithMutateFunc(testf.Transform(`.spec.components.kserve.serving.ingressGateway.certificate |= del(.secretName)`)),
+	)
+	tc.verifyKserveStatus(t, metav1.ConditionTrue)
+
+	t.Log("Deleting custom secret")
+	tc.DeleteResource(
+		WithMinimalObject(
+			gvk.Secret,
+			types.NamespacedName{Namespace: originalDSCI.Spec.ServiceMesh.ControlPlane.Namespace, Name: customSecretName},
+		),
+		WithWaitForDeletion(true),
+	)
+
+	t.Log("Custom certificate creation test completed successfully")
+}
+
+// ValidateInvalidCustomCertificateCreation tests rejection of an invalid custom certificate for OpenshiftDefaultIngress.
+func (tc *KserveTestCtx) ValidateInvalidCustomCertificateCreation(t *testing.T) {
+	t.Helper()
+
+	originalDSCI := tc.FetchDSCInitialization()
+	invalidCustomSecretName := "&invalid-secret-name"
+
+	t.Log("Configuring Kserve with OpenshiftDefaultIngress and invalid secret")
+	tc.EnsureResourceCreatedOrPatched(
+		WithMinimalObject(gvk.DataScienceCluster, tc.DataScienceClusterNamespacedName),
+		WithMutateFunc(testf.TransformPipeline(
+			testf.Transform(`.spec.components.kserve.serving.ingressGateway.certificate.secretName = "%s"`, invalidCustomSecretName),
+		)),
+	)
+
+	t.Log("Verifying Kserve reports not ready due to invalid secret")
+	tc.verifyKserveStatus(t, metav1.ConditionFalse, "unable to create serverless serving certificate secret.*&invalid-secret-name")
+
+	t.Log("Verifying invalid secret is not created")
+	tc.EnsureResourceDoesNotExist(
+		WithMinimalObject(gvk.Secret, types.NamespacedName{
+			Namespace: originalDSCI.Spec.ServiceMesh.ControlPlane.Namespace,
+			Name:      invalidCustomSecretName,
+		}),
+	)
+
+	t.Log("Deleting invalid secretName from DSC and verifying Kserve readiness")
+	tc.EnsureResourceCreatedOrPatched(
+		WithMinimalObject(gvk.DataScienceCluster, tc.DataScienceClusterNamespacedName),
+		WithMutateFunc(testf.Transform(`.spec.components.kserve.serving.ingressGateway.certificate |= del(.secretName)`)),
+	)
+	tc.verifyKserveStatus(t, metav1.ConditionTrue)
+
+	t.Log("Invalid custom certificate creation test completed successfully")
+}
+
+func (tc *KserveTestCtx) verifyKserveStatus(t *testing.T, expectedStatus metav1.ConditionStatus, expectedMessage ...string) {
+	t.Helper()
+	jqKserveReadyTrue := `
+	.status.conditions[]
+	| select(.type == "KserveReady")
+	| .status == "True"`
+
+	jqKserveReadyFalse := `
+	.status.conditions[]
+	| select(.type == "KserveReady" and .status == "False")
+	| .message | test("%s"; "i")`
+
+	condition := jq.Match(jqKserveReadyTrue)
+	if expectedStatus == metav1.ConditionFalse {
+		if len(expectedMessage) == 0 {
+			t.Fatal("expectedMessage is required when expectedStatus is False")
+		}
+		condition = jq.Match(jqKserveReadyFalse, expectedMessage[0])
+	}
+
+	tc.EnsureResourceExists(
+		WithMinimalObject(gvk.DataScienceCluster, tc.DataScienceClusterNamespacedName),
+		WithCondition(condition),
+	)
```

Comment:
```suggestion
	if expectedStatus == metav1.ConditionFalse {
		if len(expectedMessage) == 0 {
			t.Fatal("expectedMessage is required when expectedStatus is False")
		}
		tc.EnsureResourceExists(
			WithMinimalObject(gvk.DataScienceCluster, tc.DataScienceClusterNamespacedName),
			WithCondition(jq.Match(`
				.status.conditions[]
				| select(.type == "KserveReady" and .status == "False")
				| .message | test("%s"; "i")`, expectedMessage[0])),
		)
		return
	}

	tc.EnsureResourceExists(
		WithMinimalObject(gvk.DataScienceCluster, tc.DataScienceClusterNamespacedName),
		WithCondition(jq.Match(`
			.status.conditions[]
			| select(.type == "KserveReady")
			| .status == "True"`)),
	)
```

Refactor it using the line of sight pattern: https://medium.com/@matryer/line-of-sight-in-code-186dd7cdea88.